aws/resource_aws_lambda_event_source_mapping_test.go

```diff
@@ -288,17 +379,18 @@ resource "aws_lambda_function" "lambda_function_test_update" {
 }
 
 resource "aws_lambda_event_source_mapping" "lambda_event_source_mapping_test" {
-		batch_size = 100
-		event_source_arn = "${aws_kinesis_stream.kinesis_stream_test.arn}"
-		enabled = true
-		depends_on = ["aws_iam_policy_attachment.policy_attachment_for_role"]
-		function_name = "${aws_lambda_function.lambda_function_test_create.arn}"
-		starting_position = "TRIM_HORIZON"
+    batch_size = 100
```

Comment:
Given that `batch_size` had its default removed, it should also be removed from the existing test configurations and separately tested in its own acceptance test and test configuration.

In the cases where you remove it from the test configurations, I have a hunch you will start seeing test failures with the resource wanting to change it since the API likely always returns a value, but its not specified in the configuration. e.g. `batch_size: "100" => ""`.

If that is the case, we need to add a `DiffSuppressFunc` to the `batch_size` attribute.

Reply:
@bflad Can you expand on this a bit.  I went to take a look.  but i think there's a small snag.  If i'm understanding right, we would want to supress things if ```old``` was ```100```, and ```new``` was ```<empty>```, **and** this was a Kinesis or DynamoDB case.  

However, the problem i'm having is figuring out how to tell if this is Kinesis or DynamoDB.  event_source_arn doesn't really tell me this, as it's just an arbitrary ARN.  Or is the intent that i would somehow get the true value here and try to 'sniff' it to tell what was going on?

Sorry, i've looked around a bit to see if i can figure out the right way to address this.  But it was not clear to me.  So any help would def be appreciated.  Thanks!

Reply:
Luckily the ARN has the service name in it 😄  Does something like this do the trick?

```go
				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					eventSourceARN, err := arn.Parse(d.Get("event_source_arn").(string))
					if err != nil {
						return false
					}
					switch eventSourceARN.Service {
					case dynamodb.ServiceID:
						if old == "100" && new == "" {
							return true
						}
					case kinesis.ServiceID:
						if old == "100" && new == "" {
							return true
						}
					case sqs.ServiceID:
						if old == "1" && new == "" {
							return true
						}
					}
					return false
				},
```

Reply:
Slightly cleaner version:

```go
				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					if new != "" {
						return false
					}
					eventSourceARN, err := arn.Parse(d.Get("event_source_arn").(string))
					if err != nil {
						return false
					}
					if eventSourceARN.Service == dynamodb.ServiceID && old == "100" {
						return true
					}
					if eventSourceARN.Service == kinesis.ServiceID && old == "100" {
						return true
					}
					if eventSourceARN.Service == sqs.ServiceID && old == "1" {
						return true
					}
					return false
				},
```

Reply:
could have also done similar with multiple case statements:
```go
				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					eventSourceARN, err := arn.Parse(d.Get("event_source_arn").(string))
					if err != nil {
						return false
					}
					switch eventSourceARN.Service {
					case dynamodb.ServiceID, kinesis.ServiceID:
						if old == "100" && new == "" {
							return true
						}
					case sqs.ServiceID:
						if old == "1" && new == "" {
							return true
						}
					}
					return false
				},
```

Reply:
Thanks guys.  That looks like it will do the trick.  What i was missing was things like arn.Parse and kinesis.ServiceID.  Much better than me manually groveling the string!

Reply:
Ok.  Was able to make this work.  Note: i needed to use .ServiceName not .ServiceID.  Also, i needed to check for hte "0" value, not the "" value as it looks like that's what you get by default for integral values...